File: xchanger/src/main/java/com/finsim/xchanger/model/Stock.java
package com.finsim.xchanger.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.mongodb.lang.Nullable;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "stocks")
public class Stock {
    @Id
    public String isin;

    public String symbol;
    public String description;
    public String type;

    @Nullable
    public StockPrices prices;

    @Nullable
    public Order[] sellOrders;
    @Nullable
    public Order[] buyOrders;
}

File: mainframe-lib/xchanger/model/stock.go
package model

type Stock struct {
	Isin        string `json:"isin"`
	Symbol      string `json:"symbol"`
	Description string `json:"description"`
	Type        string `json:"type"`
	Prices      struct {
		DailyMax     Price `json:"dailyMax"`
		DailyMin     Price `json:"dailyMin"`
		DailyOpening Price `json:"dailyOpening"`
	} `json:"prices"`
	SellOrders []Order `json:"sellOrders,omitempty"`
	BuyOrders  []Order `json:"buyOrders,omitempty"`
}

type InsertStockInput struct {
	Isin        string `json:"isin"`
	Symbol      string `json:"symbol"`
	Description string `json:"description"`
	Type        string `json:"type"`
}
